EIPS/eip-draft_multiple_contenthash_records.md

```diff
@@ -0,0 +1,79 @@
+
+  ```solidity
+  function setContenthash(bytes32 node, bytes calldata proto, bytes calldata hash) external authorised(node);
+  ```
+
+* For getting a `contenthash` record, the `contenthash` MUST provide additional `proto` parameter and use it to get the `contenthash` for requested type. When `proto` is not provided, it MUST return the default record.
+
+  ```solidity
+  function contenthash(bytes32 node, bytes calldata proto) external view returns (bytes memory);
+  ```
+
+* Resolver that supports multiple `contenthash` records MUST return `true` for `supportsInterface` with interface ID `0x6de03e07`.
+
+Applications that are using ENS `contenthash` records SHOULD handle them in a specific way:
+
+* If the application only supports one hosting system (like directly handling ENS from IPFS/Swarm gateways), it SHOULD request `contenthash` with a specific type. The contract MUST then return it and application SHOULD correctly handle it.
+
+* If the application supports multiple hosting systems (like MetaMask), it SHOULD request `contenthash` without a specific type (like in EIP 1577). The contract MUST then return the default `contenthash` record.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+The proposed implementation was chosen because it is simple to implement and supports all important requested features. However, it doesn't support multiple records for the same type and priority order, as they aren't so important and are harder to implement properly. To implement them, a different way of implementation would need to be used.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+
+The EIP is backwards-compatible with EIP 1577, the only difference is additional overloaded methods. Old applications will still be able to function correctly, as they will receive the default `contenthash` record.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+
+Automatic Truffle tests have also been written along with new contract code. However, additional manual tests are also needed.
```

Comment:
Tests should either be included here (inline or as an asset in `../assets/eip-2520/`) or this section should just be left out entirely.

An ideal test case for an EIP just defines input and expected output pairs, but does not the test infrastructure or specific test tooling.